source/puppet/4.10/lang_data_number.md

```diff
@@ -16,38 +15,38 @@ You can work with numbers using Puppet's [arithmetic operators.][arithmetic]
 
 Numbers are written without quotation marks, and can consist only of:
 
-* Digits
-* An optional negative sign (`-`; this is actually [the unary negation operator](./lang_expressions.html#subtraction-and-negation) rather than part of the number)
-    * Explicit positive signs (`+`) aren't allowed, since there's no unary `+` operator.
-* An optional decimal point (which results in a floating point value)
-* A prefix, for octal or hexidecimal bases
-* An optional `e` or `E` for scientific notation of floating point values
+-   Digits.
+-   An optional negative sign (`-`). This is the [unary negation operator](./lang_expressions.html#subtraction-and-negation), rather than part of the number.
+    -   Explicit positive signs (`+`) aren't allowed, because there's no unary `+` operator.
+-   An optional decimal point, which results in a floating point value.
+-   A prefix, for octal or hexidecimal bases.
+-   An optional `e` or `E` for scientific notation of floating point values.
 
 ### Integers
 
 Integers are numbers without decimal points.
 
-If you divide two integers, the result _will not_ be a float; instead, Puppet will truncate the remainder. (That is, `2/3 == 0`.)
+If you divide two integers, the result _is not_ a floating point number. Instead, Puppet truncates the remainder. For example, `2/3 == 0`.
 
 ### Floating point numbers
 
-Floating point numbers ("floats") are numbers that include a fractional value after a decimal point (even if that fractional value is zero, like `2.0`).
+Floating point numbers ("floats") are numbers that include a fractional value after a decimal point, even if that fractional value is zero (such as `2.0`).
```

Comment:
A number with an exponent (scientific notation) such as 2e32 is also a float even if it doesn't have decimal point. E.g. the following are all legal floats
```
1.2
12e-1
12.32e1
-12.32e+2
```